addon/TabTracker.js

```diff
@@ -177,7 +178,7 @@ TabTracker.prototype = {
       return;
     }
     if (this._tabData.start_time) {
-      this._tabData.session_duration = (Date.now() - this._tabData.start_time);
+      this._tabData.session_duration = (absPerf.now() - this._tabData.start_time);
```

Comment:
I'd recommend to expose two APIs for absPerf, i.e. one could be used as a drop-in replacement of `Date.now()` which returns an integer, and the other one returns a floatpoint value that has the microsecond resolution.